pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,448 @@
+""" Functions to access data from the Solcast API.
+"""
+
+import requests
+import pandas as pd
+from dataclasses import dataclass
+
+
+BASE_URL = "https://api.solcast.com.au/data"
+
+
+@dataclass
+class ParameterMap:
+    solcast_name: str
+    pvlib_name: str
+    conversion: callable = lambda x: x
+
+
+# define the conventions between Solcast and PVLib nomenclature and units
+VARIABLE_MAP = [
+    # air_temp -> temp_air (deg C)
+    ParameterMap("air_temp", "temp_air"),
+    # surface_pressure (hPa) -> pressure (Pa)
+    ParameterMap("surface_pressure", "pressure", lambda x: x*100),
+    # dewpoint_temp -> temp_dew (deg C)
+    ParameterMap("dewpoint_temp", "temp_dew"),
+    # gti (W/m^2) -> poa_global (W/m^2)
+    ParameterMap("gti", "poa_global"),
+    # wind_speed_10m (m/s) -> wind_speed (m/s)
+    ParameterMap("wind_speed_10m", "wind_speed"),
+    # wind_direction_10m (deg) -> wind_direction  (deg)
+    ParameterMap("wind_direction_10m", "wind_direction"),
+    # azimuth -> solar_azimuth (degrees) (different convention)
+    ParameterMap(
+        "azimuth", "solar_azimuth", lambda x: -x % 360
+    ),
+    # precipitable_water (kg/m2) -> precipitable_water (cm)
+    ParameterMap("precipitable_water", "precipitable_water", lambda x: x*10),
+    # zenith -> solar_zenith
+    ParameterMap("zenith", "solar_zenith")
+]

```

Comment:
@AdamRJensen This is an interesting approach that we might consider adopting in some centralized way for iotools in general.

No action needed in this PR, of course.

Reply:
This is pretty cool